lib/effects/sequence_effect.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:meta/meta.dart';
+
+import './effects.dart';
+import '../position.dart';
+
+class SequenceEffect extends PositionComponentEffect {
+  final List<PositionComponentEffect> effects;
+  int _currentIndex = 0;
+  PositionComponentEffect currentEffect;
+  bool _currentWasAlternating;
+  double _driftModifier = 0.0;
+
+  SequenceEffect({
+    @required this.effects,
+    isInfinite = false,
+    isAlternating = false,
+  }) : super(isInfinite, isAlternating);
+
+  @override
+  set component(_comp) {
+    super.component = _comp;
+    final originalSize = _comp.toSize();
+    final originalPosition = _comp.toPosition();
+    Position currentSize = _comp.toSize();
+    Position currentPosition = _comp.toPosition();
+    effects.forEach((effect) {
+      effect.reset();
+      _comp.setBySize(currentSize);
+      _comp.setByPosition(currentPosition);
+      effect.component = _comp;
+      if (effect is MoveEffect) {
+        currentPosition = effect.destination;
+      } else if (effect is ScaleEffect) {
+        currentSize = Position.fromSize(effect.size);
+      }
+    });
+    travelTime = effects.fold(
+        0,
+        (time, effect) =>
+            time + effect.travelTime * (effect.isAlternating ? 2 : 1));
```

Comment:
trailing comma :)

Reply:
I think the base PositionComponentEffect would benefit from a totalTravelTime getter `= travelTime * (isAlternating ? 2 : 1)` with a nice comment distinguishing it from `travelTime`, what do you think?

Reply:
That is a good idea. :)

Reply:
Trailing comma for the fold by the way? 
Isn't that very strange syntax since it will never have any more arguments?